dashboard/app/models/script.rb

```diff
@@ -367,34 +369,52 @@ def self.get_without_cache(id_or_name)
     unless is_id
       # We didn't find a script matching id_or_name. Next, look for a script
       # in the id_or_name script family to redirect to, e.g. csp1 --> csp1-2017.
-      script_with_redirect = Script.get_script_family_redirect(id_or_name)
+      script_with_redirect = Script.get_script_family_redirect(id_or_name, version_year: version_year)
       return script_with_redirect if script_with_redirect
     end
 
-    raise ActiveRecord::RecordNotFound.new("Couldn't find Script with id|name=#{id_or_name}")
+    raise ActiveRecord::RecordNotFound.new("Couldn't find Script with id|name|family_name=#{id_or_name}")
   end
 
-  def self.get_from_cache(id_or_name)
-    return get_without_cache(id_or_name) unless should_cache?
-
-    script_cache.fetch(id_or_name.to_s) do
+  # Returns the script with the specified id, or a script with the specified
+  # name, or a redirect to the latest version of the script within the specified
+  # family. Also populates the script cache so that future responses will be cached.
+  # For example:
+  #   get_from_cache('11') --> script_cache['11'] = <Script id=11, name=...>
+  #   get_from_cache('frozen') --> script_cache['frozen'] = <Script name="frozen", id=...>
+  #   get_from_cache('csp1') --> script_cache['csp1'] = <Script redirect_to="csp1-2018">
+  #   get_from_cache('csp1', version_year: '2017') --> script_cache['csp1/2017'] = <Script redirect_to="csp1-2017">
```

Comment:
`Script.get_from_cache` now does a bit more, described here. Not mentioned in comments is that the first two examples are pre-populated by `script_cache_from_db`, while the last two are not. 

I chose this option as the shortest path. If it seems like this leaves `script_cache` too overloaded, I could separate out a new script_family_cache. Doing this would probably require moving the 404 logic from `get_without_cache` to `get_from_cache`.

Reply:
This seems reasonable to me, everything in `script_cache` is still a `Script` instance. `id_or_name` might not be the most accurate param name, but I can't think of anything better (`id_or_name_or_family_name` doesn't quite roll off the tongue).